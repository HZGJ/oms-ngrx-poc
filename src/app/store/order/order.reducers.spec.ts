import * as store from ".";

const actions = store.orderActions;
const effects = store.orderEffects;
const reducers = store.orderReducers;
const selectors = store.orderSelectors;

describe("Order Reducer", () => {
  describe("Unknown Action", () => {
    it("should return the intial state", () => {
      const action = {} as any;
      const result = reducers.orderReducer(reducers.initialState, action);

      expect(result).toBe(reducers.initialState);
    });
  });

  describe("LoadLegacy", () => {
    it("should return the initalState", () => {
      const order: store.Order = {
        id: "a1",
        name: "test",
        lineItems: null
      };

      const action = new actions.LoadLegacy();
      const result: any = reducers.orderReducer(reducers.initialState, action);

      expect(result.loading).toEqual(true);
      expect(result.loaded).toEqual(undefined);
      expect(result.entities).toMatchSnapshot();
    });
  });

  describe("LoadLegacySuccess", () => {
    it("should return the updated order entities", () => {
      const order: store.Order = {
        id: "a1",
        name: "test",
        lineItems: null
      };

      const action = new actions.LoadLegacySuccess(order);
      const result: any = reducers.orderReducer(reducers.initialState, action);

      expect(result.loading).toEqual(false);
      expect(result.loaded).toEqual(true);
      expect(result.entities).toMatchSnapshot();
    });
  });
});
