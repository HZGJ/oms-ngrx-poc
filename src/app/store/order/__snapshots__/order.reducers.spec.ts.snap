// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Reducer LoadLegacy should return the initalState 1`] = `Object {}`;

exports[`Order Reducer LoadLegacySuccess should return the updated order entities 1`] = `
Object {
  "a1": Object {
    "id": "a1",
    "lineItems": null,
    "name": "test",
  },
}
`;
